### ISSUE TITLE
IRC reconnects failing

### ISSUE THREAD

**Issue by 42wim** [42wim/matterbridge#1486]
Same happens on IRC. Even though logs claim that it's reconnecting, it's not.

Since 2021-03-09T03:52:16Z the log is full of the following.

<details>
<summary>log</summary>

```
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw (ID: #openmwirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-cs (ID: #openmw-csirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-funposting (ID: #openmw-funpostingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-offtopic (ID: #openmw-offtopicirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-troubleshooting (ID: #openmw-troubleshootingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-development (ID: #openmw-developmentirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-modding (ID: #openmw-moddingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-russian (ID: #openmw-russianirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-screenshots (ID: #openmw-screenshotsirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-development (ID: #openmw-developmentirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-modding (ID: #openmw-moddingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-russian (ID: #openmw-russianirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-screenshots (ID: #openmw-screenshotsirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-troubleshooting (ID: #openmw-troubleshootingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw (ID: #openmwirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-cs (ID: #openmw-csirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-funposting (ID: #openmw-funpostingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-offtopic (ID: #openmw-offtopicirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-shaders (ID: #openmw-shadersirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-development (ID: #openmw-developmentirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-modding (ID: #openmw-moddingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-russian (ID: #openmw-russianirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-screenshots (ID: #openmw-screenshotsirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-troubleshooting (ID: #openmw-troubleshootingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw (ID: #openmwirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-cs (ID: #openmw-csirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-funposting (ID: #openmw-funpostingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-offtopic (ID: #openmw-offtopicirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-shaders (ID: #openmw-shadersirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-funposting (ID: #openmw-funpostingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-offtopic (ID: #openmw-offtopicirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-shaders (ID: #openmw-shadersirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw (ID: #openmwirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-cs (ID: #openmw-csirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-russian (ID: #openmw-russianirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-screenshots (ID: #openmw-screenshotsirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-troubleshooting (ID: #openmw-troubleshootingirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-development (ID: #openmw-developmentirc.openmw)" prefix=irc
time="2021-03-09T04:43:36Z" level=info msg="irc.openmw: joining #openmw-modding (ID: #openmw-moddingirc.openmw)" prefix=irc
time="2021-03-09T04:43:41Z" level=error msg="disconnect: error: EOF" prefix=irc
time="2021-03-09T04:43:41Z" level=info msg="reconnecting in 30 seconds..." prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw (ID: #openmwirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-cs (ID: #openmw-csirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-funposting (ID: #openmw-funpostingirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-offtopic (ID: #openmw-offtopicirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-shaders (ID: #openmw-shadersirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-development (ID: #openmw-developmentirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-modding (ID: #openmw-moddingirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-russian (ID: #openmw-russianirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-screenshots (ID: #openmw-screenshotsirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-troubleshooting (ID: #openmw-troubleshootingirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-modding (ID: #openmw-moddingirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-russian (ID: #openmw-russianirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-screenshots (ID: #openmw-screenshotsirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-troubleshooting (ID: #openmw-troubleshootingirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-development (ID: #openmw-developmentirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-cs (ID: #openmw-csirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-funposting (ID: #openmw-funpostingirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-offtopic (ID: #openmw-offtopicirc.openmw)" prefix=irc
time="2021-03-09T04:44:19Z" level=info msg="irc.openmw: joining #openmw-shaders (ID: #openmw-shadersirc.openmw)" prefix=irc
```

</details>

The pair 
```
time="2021-03-09T03:52:21Z" level=error msg="disconnect: error: EOF" prefix=irc
time="2021-03-09T03:52:21Z" level=info msg="reconnecting in 30 seconds..." prefix=irc
```
is happening every minutes or so, but I never see the bot on IRC after the server died.

_Originally posted by @DarthGandalf in https://github.com/42wim/matterbridge/issues/1419#issuecomment-793647964_

**Comment by 42wim** [comment]
@darthgandalf

That are a lot of channels, maybe it keeps getting disconnected because of channel join flooding.
Can you run with `--debug` and add `DebugLevel=2` under your `[irc.xxx]` header?

**Comment by DarthGandalf** [comment]
I don't think it's about join flooding. If I notice that it stopped working, I go and simply restart the process, then it works again. Every time this happened, a restart helped.

And I saw once that it did reconnect to server, but didn't relay anything after that. I don't know which exact version it was.

Could it be some race with re-registering hooks? I never looked at matterbridge internals, so don't know what I'm talking about, sorry.

> Can you run with --debug and add DebugLevel=2 under your [irc.xxx] header?

Right now I'm not using matterbridge anymore, this issue being one of reasons (another reason is absence of puppets on matrix side - relaybot looks ugly). So I can't really test this without destroying the existing another bridge first (without destroying it, we'd have a loop)

The current solution we use is https://github.com/reactiflux/discord-irc for IRC/Discord, and https://t2bot.io for Discord/Matrix. Such sequential bridging has its own issues, but at least it handles server restarts properly...

**Comment by 42wim** [comment]
Ok, I didn't know you were using an outdated version.
There has been irc fixes for exactly your issue in v1.20.0

**Comment by DarthGandalf** [comment]
I was using whatever version is available from docker, and I update it whenever a new version is pushed to hub. But thanks for fix!
